apiserver/debuglog.go

```diff
@@ -182,29 +182,37 @@ func (s *debugLogSocketImpl) sendLogRecord(record *params.LogMessage) error {
 
 // debugLogParams contains the parsed debuglog API request parameters.
 type debugLogParams struct {
-	startTime     time.Time
-	maxLines      uint
-	fromTheStart  bool
-	noTail        bool
-	backlog       uint
-	filterLevel   loggo.Level
-	includeEntity []string
-	excludeEntity []string
-	includeModule []string
-	excludeModule []string
-	includeLabel  []string
-	excludeLabel  []string
+	startTime      time.Time
+	fromTheStart   bool
+	noTail         bool
+	latestLogCount uint
+	filterLevel    loggo.Level
+	includeEntity  []string
+	excludeEntity  []string
+	includeModule  []string
+	excludeModule  []string
+	includeLabel   []string
+	excludeLabel   []string
 }
 
 func readDebugLogParams(queryMap url.Values) (debugLogParams, error) {
 	var params debugLogParams
 
+	if value := queryMap.Get("backlog"); value != "" {
```

Comment:
i reordered it intentionally because backlog has a default value of 10, so if we were to check limit first, and then backlog, this will cause LatestLogCount to be overriden

Reply:
ok